### ISSUE TITLE
Use hover as fallback on desktop platforms

### ISSUE THREAD

**Issue by schalkneethling** [mdn/yari#2439]
While we _need_ `onClick` on mobile devices for expanding and collapsing the main menu on mobile devices and tablets, we can have `hover` as a fallback on desktop when the JavaScript code fails.

Here is how I foresee the implementation.

- Add a block to the main menu CSS that is scoped to the class `nojs`.
- Inside this block add the relevant styling that will allow the menu to open and close on hover
- Update the React JavaScript code to remove the `nojs` class when attaching its event handler

This way a user can still expand and interact with the main menu if there is an error that causes the JavaScript to fail but, if all is well, the menu interaction will be as it is currently implemented.

Part of https://github.com/mdn/yari/issues/1124

**Comment by schalkneethling** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Use hover as fallback on desktop platforms](https://trello.com/c/9bqDWZXW/20-use-hover-as-fallback-on-desktop-platforms)

**Comment by mattiapontonio** [comment]
There is not `hover` in mobile.
There is `click`.

**Comment by tannerdolby** [comment]
@mattiapontonio That's not necessarily true. There **is** `hover` in mobile depending on the device (iOS or Android). If an element is "clickable" then it's `hover` state can be activated by "clicking" on it but the event is not explicitly a `click` event. 

Also, in iOS, there is something called [Touch Events](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Using_Touch_Events) that introduce some new fun. 

This being said. I think using `hover` as a fallback on desktop platforms when JS fails is a great fix.
